# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsep/JsepSessionImpl.cpp
# Commit: d570e7369906
# Full Hash: d570e7369906010743035437ff6ff4b4c7a34e67
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2022-04-06 04:35:13
# Regressor Bug: 1624989
# File Overlap Count: 1
# Description:
#   Bug 1624989: Move RTCPeerConnection's operations chain to c++ r=jib,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D141738
# ==============================================================================

diff -r 056457fa52fe -r d570e7369906 dom/media/webrtc/jsep/JsepSessionImpl.cpp
--- a/dom/media/webrtc/jsep/JsepSessionImpl.cpp	Tue Apr 05 17:41:37 2022 +0000
+++ b/dom/media/webrtc/jsep/JsepSessionImpl.cpp	Tue Apr 05 18:39:12 2022 +0000
@@ -88,6 +88,31 @@
   return NS_OK;
 }
 
+static void GetIceCredentials(
+    const Sdp& aSdp,
+    std::set<std::pair<std::string, std::string>>* aCredentials) {
+  for (size_t i = 0; i < aSdp.GetMediaSectionCount(); ++i) {
+    const SdpAttributeList& attrs = aSdp.GetMediaSection(i).GetAttributeList();
+    if (attrs.HasAttribute(SdpAttribute::kIceUfragAttribute) &&
+        attrs.HasAttribute(SdpAttribute::kIcePwdAttribute)) {
+      aCredentials->insert(
+          std::make_pair(attrs.GetIceUfrag(), attrs.GetIcePwd()));
+    }
+  }
+}
+
+std::set<std::pair<std::string, std::string>>
+JsepSessionImpl::GetLocalIceCredentials() const {
+  std::set<std::pair<std::string, std::string>> result;
+  if (mCurrentLocalDescription) {
+    GetIceCredentials(*mCurrentLocalDescription, &result);
+  }
+  if (mPendingLocalDescription) {
+    GetIceCredentials(*mPendingLocalDescription, &result);
+  }
+  return result;
+}
+
 nsresult JsepSessionImpl::AddTransceiver(RefPtr<JsepTransceiver> transceiver) {
   mLastError.clear();
   MOZ_MTLOG(ML_DEBUG, "[" << mName << "]: Adding transceiver.");
@@ -2365,7 +2390,7 @@
 
 const std::string JsepSessionImpl::GetLastError() const { return mLastError; }
 
-const std::vector<std::pair<size_t, std::string> >&
+const std::vector<std::pair<size_t, std::string>>&
 JsepSessionImpl::GetLastSdpParsingErrors() const {
   return mLastSdpParsingErrors;
 }